lib/Jsrt/Jsrt.cpp

```diff
@@ -43,26 +43,28 @@ struct CodexHeapAllocatorInterface
 typedef utf8::NarrowWideConverter<CodexHeapAllocatorInterface, LPCSTR, LPWSTR> NarrowToWideChakraHeap;
 typedef utf8::NarrowWideConverter<CodexHeapAllocatorInterface, LPCWSTR, LPSTR> WideToNarrowChakraHeap;
 
-JsErrorCode CheckContext(JsrtContext *currentContext, bool verifyRuntimeState, bool allowInObjectBeforeCollectCallback)
+JsErrorCode CheckContext(JsrtContext *currentContext, bool verifyRuntimeState,
+    bool allowInObjectBeforeCollectCallback)
 {
     if (currentContext == nullptr)
     {
         return JsErrorNoCurrentContext;
     }
 
-    Js::ScriptContext *scriptContext = currentContext->GetScriptContext();
-    Assert(scriptContext != nullptr);
-    Recycler *recycler = scriptContext->GetRecycler();
-    ThreadContext *threadContext = scriptContext->GetThreadContext();
-
     // We don't need parameter check if it's checked in previous wrapper.
     if (verifyRuntimeState)
     {
+        Js::ScriptContext *scriptContext = currentContext->GetScriptContext();
+        Assert(scriptContext != nullptr);
+        Recycler *recycler = scriptContext->GetRecycler();
+        ThreadContext *threadContext = scriptContext->GetThreadContext();
+
         if (recycler && recycler->IsHeapEnumInProgress())
         {
             return JsErrorHeapEnumInProgress;
         }
-        else if (!allowInObjectBeforeCollectCallback && recycler && recycler->IsInObjectBeforeCollectCallback())
+        else if (!allowInObjectBeforeCollectCallback &&
+            recycler && recycler->IsInObjectBeforeCollectCallback())
```

Comment:
>IsInObjectBeforeCollectCallback [](start = 34, length = 31)

we don't get this information from `JsHasException`. How can host determine that?